pydantic/typing.py

```diff
@@ -363,6 +368,38 @@ def is_typeddict(type_: Type[Any]) -> bool:
     return lenient_issubclass(type_, dict) and hasattr(type_, '__total__')
 
 
+def is_typeddict_special(type_: Type[Any]) -> bool:
+    """
+    Check if type is a TypedDict special form (Required or NotRequired).
+    """
+    return type_ is TypedDictRequired or type_ is TypedDictNotRequired
```

Comment:
```suggestion
def is_typeddict_special(type_: Any) -> bool:
    """
    Check if type is a TypedDict special form (Required or NotRequired).
    """
    return type_ is not None and (type_ is TypedDictRequired or type_ is TypedDictNotRequired)
```

maybe? would make the usage above slightly cleaner.